diff --git a/CMakeLists.txt b/CMakeLists.txt
index bab7661..9972fa5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,7 @@ set_package_properties(LibLZMA PROPERTIES
 )
 
 find_package(PkgConfig)
-pkg_check_modules(LibZstd IMPORTED_TARGET "libzstd")
+#pkg_check_modules(LibZstd IMPORTED_TARGET "libzstd")
 add_feature_info(LibZstd LibZstd_FOUND
                 "Support for zstd compressed files and data streams"
 )
