diff --git a/qtbase/src/network/ssl/ssl.pri b/qtbase/src/network/ssl/ssl.pri
index 230c45c26f..242bdacada 100644
--- a/qtbase/src/network/ssl/ssl.pri
+++ b/qtbase/src/network/ssl/ssl.pri
@@ -118,7 +118,7 @@ qtConfig(ssl) {
         qtConfig(openssl-linked): {
             android {
                 build_pass|single_android_abi: LIBS_PRIVATE += -lssl_$${QT_ARCH} -lcrypto_$${QT_ARCH}
-            } else: QMAKE_USE_FOR_PRIVATE += openssl
+            } else: LIBS_PRIVATE += /usr/local/lib/libssl.a /usr/local/lib/libcrypto.a -ldl -lpthread
         } else: \
             QMAKE_USE_FOR_PRIVATE += openssl/nolink
         win32 {
